src/run/run.c

```diff
@@ -2154,57 +2221,42 @@ static int start_transient_service(sd_bus *bus) {
 
         if (w) {
-                r = bus_wait_for_jobs_one(w,
-                                          object,
-                                          arg_quiet ? 0 : BUS_WAIT_JOBS_LOG_ERROR,
-                                          arg_runtime_scope == RUNTIME_SCOPE_USER ? STRV_MAKE_CONST("--user") : NULL);
+                r = bus_wait_for_jobs_one(
+                                w,
+                                object,
+                                arg_quiet ? 0 : BUS_WAIT_JOBS_LOG_ERROR,
+                                arg_runtime_scope == RUNTIME_SCOPE_USER ? STRV_MAKE_CONST("--user") : NULL);
                 if (r < 0)
                         return r;
-        } else if (job_done_context.match) {
-                job_done_context.start_job = strdup(object);
-                if (!job_done_context.start_job)
+        } else if (c.match_job_removed) {
+                c.start_job = strdup(object);
+                if (!c.start_job)
                         return log_oom();
         }
 
         if (!arg_quiet) {
                 sd_id128_t invocation_id;
 
-                r = acquire_invocation_id(bus, service, &invocation_id);
+                r = acquire_invocation_id(bus, c.unit, &invocation_id);
                 if (r < 0)
                         return r;
 
-                r = print_unit_invocation(service, invocation_id);
+                r = print_unit_invocation(c.unit, invocation_id);
                 if (r < 0)
                         return r;
         }
 
         if (arg_wait || arg_stdio != ARG_STDIO_NONE) {
-                _cleanup_(run_context_done) RunContext c = {
-                        .cpu_usage_nsec = NSEC_INFINITY,
-                        .memory_peak = UINT64_MAX,
-                        .memory_swap_peak = UINT64_MAX,
-                        .ip_ingress_bytes = UINT64_MAX,
-                        .ip_egress_bytes = UINT64_MAX,
-                        .io_read_bytes = UINT64_MAX,
-                        .io_write_bytes = UINT64_MAX,
-                        .inactive_exit_usec = USEC_INFINITY,
-                        .inactive_enter_usec = USEC_INFINITY,
-                };
+                c.bus_path = unit_dbus_path_from_name(c.unit);
+                if (!c.bus_path)
+                        return log_oom();
 
                 r = sd_event_default(&c.event);
                 if (r < 0)
                         return log_error_errno(r, "Failed to get event loop: %m");
 
-                c.service = strdup(service);
-                if (!c.service)
-                        return log_oom();
-
-                c.bus_path = unit_dbus_path_from_name(service);
-                if (!c.bus_path)
-                        return log_oom();
-
                 _cleanup_(osc_context_closep) sd_id128_t osc_context_id = SD_ID128_NULL;
-                if (pty_fd >= 0) {
+                if (c.pty_fd >= 0) {
                         if (arg_exec_user && !terminal_is_dumb()) {
```

Comment:
Hmm, maybe move the osc context setup to run_context_setup_ptyfwd too?

Reply:
but then we'd have to pass through the osc_context_id field, and it isn't stricty about ptyfwd stuff anyway. dunno.